### PR TITLE
PDCL-4210 - fixing the click storage not to have duplicates

### PR DESCRIPTION
**Pull request by ninaceban**
## Description

In this PR I'm fixing 2 bugs:
- @Aaronius discovered that at `viewChange` we will collect duplicates of the clicks and then include those duplicates in the `meta` sent to konductor
- While manual testing it I have discovered that when having an SPA and same element in both of the pages, while navigating from one page to another we'll have again duplicates of clicks but this time for diferent `scope`
This is happening because a `click` metric will be collected only when we'll render a decision. At `viewChange` we will collect clicks for that specific view. Now if we navigate from Home -> Consent -> Links -> Consent we will have the Consent metrics duplicated. Also if Consent and Links has an `h2` element right next to the body tag, VEC will compute the same selector for both of the views. This means that whenever we'll click on it while on Links view, the conversion will increase for both Links and Consent metrics for that specific selector. This is an edge case because it will happen only when specific navigation occurs but still I don't consider that Is quite right because our reports will be inconsistent.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by jfkhoury** [comment]
Thank you so much for the detailed description @ninaceban

File: src/components/Personalization/dom-actions/clicks/collectClicks.js [hunk 1]
```diff
@@ -34,12 +41,16 @@ export default (clickedElement, values) => {
   }
```
Comment (jonsnyder) [inline src/components/Personalization/dom-actions/clicks/collectClicks.js:41]: I think this would be more clear if you renamed `values` to `clickStorage`

**Review by jonsnyder — APPROVED** [review]

**Review by Aaronius — APPROVED** [review]
Thanks Nina! I left a comment that I think should be addressed. If you choose not to address it now because you'd like to get these changes into the release, please create a JIRA ticket so that we can address it soon.

File: src/components/Personalization/index.js [hunk 2]
```diff
@@ -27,13 +27,13 @@ import createViewCacheManager from "./createViewCacheManager";
 import createViewChangeHandler from "./createViewChangeHandler";
 import decisionsExtractor from "./decisionsExtractor";
 import createOnResponseHandler from "./createOnResponseHandler";
+import createClickStorage from "./createClickStorage";
 
 const createPersonalization = ({ config, logger, eventManager }) => {
   const collect = createCollect({ eventManager, mergeMeta });
   const viewCollect = createViewCollect({ eventManager, mergeMeta });
-  const clickStorage = [];
+  const { clickStorage, store } = createClickStorage();
```
[the remark this answers is not in the document]
Comment (ninaceban) [reply src/components/Personalization/index.js]: thank you @Aaronius

**Comment by ninaceban** [comment]
@Aaronius and @jonsnyder could you please take a look at the updates! thank you for your reviews

**Comment by Aaronius** [comment]
Looks good to me. Thanks Nina!

**Review by jonsnyder — APPROVED** [review]

**Pull request merged**